trailing_spaces.py

```diff
@@ -90,19 +127,28 @@ def find_trailing_spaces(view):
     include_current_line = bool(ts_settings.get("trailing_spaces_include_current_line",
                                                 DEFAULT_IS_ENABLED))
     regexp = ts_settings.get("trailing_spaces_regexp") + "$"
-    no_empty_lines_regexp = "(?<=\\S)%s$" % regexp
 
-    offending_lines = view.find_all(regexp if include_empty_lines else no_empty_lines_regexp)
+    if not include_empty_lines:
+        regexp = "(?<=\\S)%s$" % regexp
 
-    sel = view.sel()
-    line = len(sel) and view.line(sel[0].b)
+    # find all matches in the currently visible region plus a little before and
+    # after
+    searched_region = view.visible_region()
+    searched_region.a = max(searched_region.a - trailing_spaces_non_visible_highlighting, 0);
+    searched_region.b = min(searched_region.b + trailing_spaces_non_visible_highlighting, view.size());
 
-    if include_current_line or not line:
+    searched_region = view.line(searched_region) # align to line start and end
+    offending_lines = view_find_all_in_region(view, searched_region, regexp)
+
+    if include_current_line or len(view.sel()) == 0:
         return [offending_lines, offending_lines]
     else:
-        current_offender = view.find(regexp if include_empty_lines else no_empty_lines_regexp, line.a)
-        removal = False if current_offender is None else line.intersects(current_offender)
-        highlightable = [i for i in offending_lines if i != current_offender] if removal else offending_lines
+        # find all matches in the current line and exclude them from highlighting
+        sel = view.sel()[0]
+        line = view.line(sel.b)
```

Comment:
Previous code (on master) was a bit nicer IMO (I might be biased as I wrote it though).

Reply:
I missed that your changes made this commit obsolete. I dropped it.